code/job.py

```diff
@@ -0,0 +1,69 @@
+from flask_restful import Resource, reqparse
+
+jobs = []
+
+class Job(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument(
+        'company',
+        type=str,
+        required=True,
+        help="This field cannot be left blank"
+        )
+    parser.add_argument('salary')
+    parser.add_argument('status')
+
+    def post(self, title):
+
+        if self.check_for_job(title) is not None:
+            return{'message': "A job with name '{}' already exists".format(title)}, 400
+
+        data = Job.parser.parse_args()
+
+        job = {
+            'title': title, 
+            'company': data['company'],
+            'salary': data['salary'],
+            'status': data['status']
+            }
+        jobs.append(job)
+
+        return job, 201
+
+    def check_for_job(self, title):
```

Comment:
Not a big fan of method names like this, as it doesn't really tell you what the method does. It certainly doesn't suggest that it will return a job if it finds it. I suggest something along the lines of get_existing_job is more descriptive, as it tells you what it's checking and that it will return something.

Reply:
I've deleted this method as it's no longer necessary: changing the jobs data structure to a dictionary means dict.get() can be used instead.